src/iden3comm/handlers/common.ts

```diff
@@ -136,6 +140,82 @@ export const processZeroKnowledgeProofRequests = async (
   return zkpResponses;
 };
 
+/**
+ * Processes auth proof requests.
+ *
+ * @param to - The identifier of the recipient.
+ * @param proofService - The proof service.
+ * @param opts - Additional options for processing the requests.
+ * @returns A promise that resolves to an auth proof response.
+ */
+export const processProofAuth = async (
+  to: DID,
+  proofService: IProofService,
+  opts: {
+    supportedCircuits: CircuitId[];
+    acceptProfile?: AcceptProfile;
+    challenge?: bigint;
+    skipRevocation?: boolean;
+  }
+): Promise<AuthProofResponse> => {
+  if (!opts.acceptProfile) {
+    opts.acceptProfile = defaultAcceptProfile;
+  }
+  if (!opts.skipRevocation) {
+    opts.skipRevocation = true;
+  }
+
+  switch (opts.acceptProfile.env) {
+    case MediaType.ZKPMessage:
+      if (!opts.acceptProfile.circuits) {
+        throw new Error('Circuit not specified in accept profile');
+      }
+
+      for (const circuitId of opts.acceptProfile.circuits) {
+        if (!opts.supportedCircuits.includes(circuitId as unknown as CircuitId)) {
+          throw new Error(`Circuit ${circuitId} is not supported`);
+        }
+
+        const zkpRes: ZeroKnowledgeProofAuthResponse = await proofService.generateAuthProof(
+          circuitId as unknown as CircuitId,
+          to,
+          { challenge: opts.challenge, skipRevocation: opts.skipRevocation }
+        );
+
+        const zkProofEncoded = packAuthV2Proof(
```

Comment:
circuit id can be other then auth v2, so se need a swith case here, and in case it is not auth v2 we must skip it and don't return auth v2 response.

case: profile contains authv3, I go here, I will pack auth v3 (in case it is supporter) but will return AUTHV2 method

Reply:
Done